test/addons/cpu-profiler/binding.cc

```diff
@@ -0,0 +1,42 @@
+#include "node.h"
+#include "v8.h"
+#include "v8-profiler.h"
+
+void Begin(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* const isolate = args.GetIsolate();
+  isolate->GetCpuProfiler()->StartProfiling(v8::String::NewFromUtf8(isolate, ""), true);
+}
+
+void End(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* const isolate = args.GetIsolate();
+  v8::CpuProfile* const profile = isolate->GetCpuProfiler()->StopProfiling(v8::String::NewFromUtf8(isolate, ""));
+
+  const int64_t samplesCount = profile->GetSamplesCount();
+
+  int64_t minInterval = -1;
+  int64_t maxInterval = -1;
+
+  for (int i = 1; i < samplesCount; i++) {
+    const int64_t interval = (profile->GetSampleTimestamp(i) - profile->GetSampleTimestamp(i - 1));
+
+    if (minInterval < 0 || interval < minInterval)
+      minInterval = interval;
+
+    if (maxInterval < 0 || interval > maxInterval)
+      maxInterval = interval;
+  }
+
+  // Check that samples are evenly distributed.
+  const bool ok = (samplesCount > 2) && ((maxInterval / minInterval) < 2);
+  args.GetReturnValue().Set(ok ? v8::True(isolate) : v8::False(isolate));
+}
+
+void Initialize(v8::Local<v8::Object> binding) {
+  v8::Isolate* const isolate = binding->GetIsolate();
+  binding->Set(v8::String::NewFromUtf8(isolate, "begin"),
+               v8::FunctionTemplate::New(isolate, Begin)->GetFunction());
+  binding->Set(v8::String::NewFromUtf8(isolate, "end"),
+               v8::FunctionTemplate::New(isolate, End)->GetFunction());
+}
+
+NODE_MODULE(test, Initialize)
```

Comment:
There are a bunch of style errors in this file but once #2427 lands, `make cpplint` should tell you about them.